server.js

```diff
@@ -11,24 +11,25 @@ app.use(express.urlencoded({ extended: true }));
 //Create a new message
 app.post("/messages", (req, res) => {
   let newPost = {
-    id: messages.length,
+    id: uuidv4(),
     from: req.body.from,
     text: req.body.text,
-    timeSent: dateTime,
+    timeSent: getTime(),
   };
-  if (newPost.from !== "" && newPost.text !== "") {
+
+  if (newPost.from !== undefined && newPost.text !== undefined) {
```

Comment:
In this case, the empty text is not handled. It is a bit painful that undefined and empty are two different things.
A way to make it nice without composing a cumbersome expression like this `newPost.from !== undefined && newPost.text !== undefined && newPost.from !== "" && newPost.text !== ""` is to utilize a built in ability in JS by relying on the auto value conversion to a boolean value in Boolean context such as in if condition.

We say in JS a value is truthy or falsy when based on its contents it will be evaluated to either true or false. This might seem a bit of magic and tricky but after a little practice, it comes to handy in many circumstances like the one we're dealing with now.

Here is the official documentation, it is super short:
https://developer.mozilla.org/en-US/docs/Glossary/Truthy
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

As a rule of thumbs, there is a limited number of falsy values that are always false, everything else is always truthy. So it is enough to memorise what is falsy.

After rewriting the code, this is how it would look like. You can see how much terser it became.
```
  if (newPost.form && newPost.text) {
    messages.push(newPost);
    res.json(messages);
    res.json(newPost);
  } else {
    res.status(400).json(`Please fill all the fields`);
  }
```